x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/pivot/AggregationResultUtils.java

```diff
@@ -211,17 +215,35 @@ public Object value(Aggregation agg, String fieldType) {
         }
     }
 
+    static class SingleBucketAggExtractor implements AggValueExtractor {
+        @Override
+        public Object value(Aggregation agg, String fieldType, Map<String, String> fieldTypeMap) {
+            SingleBucketAggregation aggregation = (SingleBucketAggregation) agg;
+
+            if (aggregation.getAggregations().iterator().hasNext() == false) {
+                return aggregation.getDocCount();
+            }
+
+            HashMap<String, Object> nested = new HashMap<>();
+            for (Aggregation subAgg : aggregation.getAggregations()) {
+                nested.put(subAgg.getName(), getExtractor(subAgg).value(subAgg, fieldTypeMap.get(agg.getName()), fieldTypeMap));
```

Comment:
```suggestion
                nested.put(subAgg.getName(), getExtractor(subAgg).value(subAgg, fieldTypeMap.get(subAgg.getName()), fieldTypeMap));
```

We should use the `subAgg` field type no? If we just used the parent agg's type, we could pass `fieldType`

Reply:
🤔 

Does this actually work? It seems like we should recursively supply the `String fieldName` or something and then read that from `fieldTypeMap`. 

That way we can continually append the field name. 

Something like
```
nested.put(subAgg.getName(), getExtractor(subAgg).value(subAgg, fieldName + "." + subAgg.getName(), fieldTypeMap));
```

And those that worry about `fieldType` can get it from the `fieldTypeMap`

Reply:
good catch! I think that makes sense.

In order to fix this, I will create a test combines both mapping deduction and value extraction, both are tested individually at the moment which makes it hard to discover bugs like this.